### PR TITLE
Remove pattern

### PR DESCRIPTION
**Pull request by m-kuhn**
Doesn't match ids like https://orcid.org/0000-0002-0572-306X

### CODE REVIEW

**Comment by rouault** [comment]
can't we keep a pattern (I assume there must be some rules), but maybe more tolerant ?

**Comment by rouault** [comment]
I see https://support.orcid.org/hc/en-us/articles/360006897674-Structure-of-the-ORCID-Identifier that it should probably be "^[0-9X]{4}-[0-9X]{4}-[0-9X]{4}-[0-9X]{4}$"

**Comment by m-kuhn** [comment]
I was thinking about allowing any alphanumeric as last char but I've got no idea about the real rules

File: zenodo-json-schema.json [hunk 1]
```diff
@@ -609,8 +609,7 @@
           },
           "orcid": {
             "description": "ORCID identifier for creator.",
-            "type": "string",
-            "pattern": "^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{4}$"
+            "type": "string"
```
Comment (rouault) [inline zenodo-json-schema.json]: actually only the last char can be X, so,
```suggestion
            "type": "string",
            "pattern": "^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9X]$"
```

Comment (m-kuhn) [reply zenodo-json-schema.json]: is that better than [0-9A-Z]?

File: zenodo-json-schema.json [hunk 2]
```diff
@@ -610,7 +610,7 @@
           "orcid": {
             "description": "ORCID identifier for creator.",
             "type": "string",
-            "pattern": "^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{4}$"
+            "pattern": "^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9A-Z]$"
```
Comment (rouault) [inline zenodo-json-schema.json]:
```suggestion
            "pattern": "^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9X]$"
```

File: zenodo-json-schema.json (hunk 1 above)
Comment (rouault) [reply zenodo-json-schema.json]: > is that better than [0-9A-Z]?

a bit stricter: "The last character in the ORCID iD is a checksum. In accordance with ISO/IEC 7064:2003, MOD 11-2, this checksum must be "0" - "9" or "X", a capital letter X which represents the value 10"

Comment (m-kuhn) [reply zenodo-json-schema.json]: well discovered

**Pull request merged**